packages/@adobe/spectrum-css-temp/components/checkbox/skin.css

```diff
@@ -84,185 +72,225 @@ governing permissions and limitations under the License.
+      &.is-indeterminate .spectrum-Checkbox-box,
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--spectrum-checkbox-quiet-box-border-color-selected-down);
+        }
+      }
     }
   }
 
-  &:hover {
+  &.is-invalid {
+    /* Extra-specific selectors added here to handle checked state */
     .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
-    .spectrum-Checkbox-box {
+    .spectrum-Checkbox-box,
+    &.is-indeterminate .spectrum-Checkbox-box,
+    .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
       &:before {
-        border-color: var(--spectrum-checkbox-box-border-color-error-hover);
+        border-color: var(--spectrum-checkbox-box-border-color-error);
       }
     }
 
     .spectrum-Checkbox-label {
-      color: var(--spectrum-checkbox-text-color-error-hover);
+      color: var(--spectrum-checkbox-text-color-error);
     }
-  }
 
-  &:active {
-    .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
-    .spectrum-Checkbox-box {
-      &:before {
-        border-color: var(--spectrum-checkbox-box-border-color-error-down);
+    &.is-indeterminate .spectrum-Checkbox-input:focus-ring,
+    .spectrum-Checkbox-input:focus-ring {
+      & + .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--spectrum-checkbox-box-border-color-error-hover);
+        }
+      }
+
+      & ~ .spectrum-Checkbox-label {
+        color: var(--spectrum-checkbox-text-color-error-hover);
       }
     }
 
-    .spectrum-Checkbox-label {
-      color: var(--spectrum-checkbox-text-color-error-down);
+    &:hover {
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
+      .spectrum-Checkbox-box,
+      &.is-indeterminate .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--spectrum-checkbox-box-border-color-error-hover);
+        }
+      }
+
+      .spectrum-Checkbox-label {
+        color: var(--spectrum-checkbox-text-color-error-hover);
+      }
+    }
+
+    &:active {
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--spectrum-checkbox-box-border-color-error-down);
+        }
+      }
+
+      .spectrum-Checkbox-label {
+        color: var(--spectrum-checkbox-text-color-error-down);
+      }
     }
   }
 }
+
 @media (forced-colors: active) {
   .spectrum-Checkbox {
     forced-color-adjust: none;
-    --spectrum-checkbox-box-border-color: ButtonText;
+    --spectrum-alias-toggle-color-default: ButtonText;
+    --spectrum-alias-toggle-color-hover: Highlight;
+    --spectrum-alias-toggle-color-key-focus: Highlight;
+    --spectrum-alias-toggle-color-selected: Highlight;
+    --spectrum-alias-toggle-color-selected-hover: Highlight;
+    --spectrum-alias-toggle-color-selected-key-focus: Highlight;
+    --spectrum-alias-toggle-color-selected-down: Highlight;
+    --spectrum-accent-color-900: Highlight;
```

Comment:
I think I should move the color variables over from the index.css file where they are currently defined:

```css
  --spectrum-checkbox-box-border-color: var(--spectrum-alias-toggle-color-default);
  --spectrum-checkbox-box-border-color-hover: var(--spectrum-alias-toggle-color-hover);
  --spectrum-checkbox-box-border-color-key-focus: var(--spectrum-alias-toggle-color-key-focus);
  --spectrum-checkbox-quiet-box-border-color-selected: var(--spectrum-alias-toggle-color-selected);
  --spectrum-checkbox-quiet-box-border-color-selected-hover: var(--spectrum-alias-toggle-color-selected-hover);
  --spectrum-checkbox-quiet-box-border-color-selected-key-focus: var(--spectrum-alias-toggle-color-selected-key-focus);

  --spectrum-checkbox-emphasized-box-background-color-selected: var(--spectrum-accent-color-900);
  --spectrum-checkbox-emphasized-box-background-color-selected-hover: var(--spectrum-accent-color-1000);
  --spectrum-checkbox-emphasized-box-background-color-selected-key-focus: var(--spectrum-accent-color-1000);
  --spectrum-checkbox-emphasized-box-background-color-selected-down: var(--spectrum-accent-color-1100);
  --spectrum-checkbox-emphasized-box-border-color-selected: var(--spectrum-accent-color-900);
  --spectrum-checkbox-emphasized-box-border-color-selected-hover: var(--spectrum-accent-color-1000);
  --spectrum-checkbox-emphasized-box-border-color-selected-key-focus: var(--spectrum-accent-color-1000);
  --spectrum-checkbox-emphasized-box-border-color-selected-down: var(--spectrum-accent-color-1100);
```

Colors should be in skin.css. 

Also, the color variables work in high contrast mode when I define the initial variables under `.spectrum-Checkbox`, rather than `:root.` Otherwise, the color variables using `var(--spectrum-accent-color-900)`, `var(--spectrum-accent-color-1000)` and `var(--spectrum-accent-color-1100)` are not getting overridden with the variables redefined under `@media (forced-colors: active)`

Reply:
haha, you beat me to explaining that

The reason for it is due to some postcss preprocessing code that tries to collapse variables. And yes, the way to get it to stop doing that is to put it in the top level component classname instead of in root.

It doesn't really matter which file the variables are declared in, but it probably makes it easier to read to keep the color ones in the skin file.